roblox_api.py

```
@@ -16,6 +16,21 @@ async def get_user_id_from_username(username: str):
+        presence_url = "https://presence.roblox.com/v1/presence/users"
+        payload = {"userIds": [userid]}
+        presence_response = requests.post(presence_url, json=payload)
+        presence_response.raise_for_status()
+        presence_data = presence_response.json()
+
+        if presence_data and presence_data["userPresences"]:
+            return presence_data["userPresences"][0]["userPresenceType"]
+        else:
+            return 0  # Assume offline if no presence data is found
+    except requests.exceptions.RequestException as e:
+        raise HTTPException(status_code=500, detail=f"Failed to fetch user presence: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)